codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker.java

```diff
@@ -143,13 +148,55 @@ protected WebSocketClientHandshaker(URI uri, WebSocketVersion version, String su
     protected WebSocketClientHandshaker(URI uri, WebSocketVersion version, String subprotocol,
                                         HttpHeaders customHeaders, int maxFramePayloadLength,
                                         long forceCloseTimeoutMillis, boolean absoluteUpgradeUrl) {
+        this(uri, version, subprotocol, customHeaders, maxFramePayloadLength, forceCloseTimeoutMillis,
+            absoluteUpgradeUrl, null, null);
+    }
+
+    /**
+     * Base constructor
+     *
+     * @param uri
+     *            URL for web socket communications. e.g "ws://myhost.com/mypath". Subsequent web socket frames will be
+     *            sent to this URL.
+     * @param version
+     *            Version of web socket specification to use to connect to the server
+     * @param subprotocol
+     *            Sub protocol request sent to the server.
+     * @param customHeaders
+     *            Map of custom headers to add to the client request
+     * @param maxFramePayloadLength
+     *            Maximum length of a frame's payload
+     * @param forceCloseTimeoutMillis
+     *            Close the connection if it was not closed by the server after timeout specified
+     * @param  absoluteUpgradeUrl
+     *            Use an absolute url for the Upgrade request, typically when connecting through an HTTP proxy over
+     *            clear HTTP
+     * @param httpRequestEncoderName
+     *            After the handshake is complete, the {@link HttpRequestEncoder} or {@link HttpClientCodec}
+     *            in the pipeline needs to be replaced with a websocket encoder.  If httpRequestEncoderName is present,
+     *            the handshaker assumes that this is the name of the http request encoder that should be replaced.
+     *            If not specified, the first {@link HttpRequestEncoder} or {@link HttpClientCodec} found in the
+     *            pipeline will be replaced.
+     * @param httpResponseDecoderName
+     *            After the handshake is complete, the {@link HttpResponseDecoder} or {@link HttpClientCodec}
+     *            in the pipeline needs to be replaced with a websocket decoder.  If httpResponseDecoderName is present,
+     *            the handshaker assumes that this is the name of the http response decoder that should be replaced.
+     *            If not specified, the first {@link HttpResponseDecoder} or {@link HttpClientCodec} found in the
+     *            pipeline will be replaced
+     */
+    protected WebSocketClientHandshaker(URI uri, WebSocketVersion version, String subprotocol,
```

Comment:
IMHO, we should not add yet another protected constructor (hence an API) and first introduce a Builder and deprecate the existing constructors.

Reply:
I think it is ok to add this as long as we make it package-private